File: src/main/kotlin/com/nazarethlabs/notes/service/note/FavoriteNoteService.kt
package com.nazarethlabs.notes.service.note

import com.nazarethlabs.notes.dto.Note
import com.nazarethlabs.notes.repository.NoteStorageRepository

class FavoriteNoteService {
    fun toggleFavorite(note: Note) {
        NoteStorageRepository
            .getInstance()
            .toggleFavorite(note.id)
    }
}

File: vscode/src/service/FavoriteNoteService.ts
import { Note } from '../dto/Note';
import { NoteRepository } from '../repository/NoteRepository';

export class FavoriteNoteService {
  private readonly repository: NoteRepository;

  constructor(repository: NoteRepository) {
    this.repository = repository;
  }

  toggleFavorite(note: Note): void {
    this.repository.toggleFavorite(note.id);
  }
}
